src/main/java/org/mule/extension/file/internal/lock/FileChannelPathLock.java

```diff
@@ -83,8 +85,8 @@ public boolean tryLock() {
    * {@inheritDoc}
    */
   @Override
-  public boolean tryLock(long timeout) {
-    long nanoTimeout = timeout < 0 ? 0 : timeout;
+  public boolean tryLock(Long timeout) {
+    long nanoTimeout = timeout < 0 ? 0 : TimeUnit.MILLISECONDS.toNanos(timeout);
```

Comment:
I would just use milliseconds everywhere

Reply:
Used nanos because it is recommended by the Java documentation of System.nanoTime(). I did not use currentTimeMillis() as it is susceptible to date and time changes in the operating system. As there is no guarantee for the starting value of System.nanoTime(), it is not convenient to convert it to milliseconds as it may convert to 0.